web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -96,14 +97,55 @@
             .map(a -> a.getAttributeId()).collect(Collectors.toList()), ClinicalDataType.PATIENT);
         combinedResult.addAll(resultForSampleAttributes);
         combinedResult.addAll(resultForPatientAttributes);
         return new ResponseEntity<>(combinedResult, HttpStatus.OK);
     }
 
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
+    @RequestMapping(value = "/oncokb/fetch", method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch annotation data by study view filter")
+    public ResponseEntity<List<OncoKBDataCountItem>> fetchOncoKBData(
+        @ApiParam(required = true, value = "Annotation data filter")
+        @Valid @RequestBody(required = false)  OncoKBDataCountFilter oncoKBDataCountFilter,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
+        @Valid @RequestAttribute(required = false, value = "interceptedOncoKBDataCountFilter") OncoKBDataCountFilter interceptedOncoKBDataCountFilter) {
+
+        List<OncoKBDataFilter> attributes = interceptedOncoKBDataCountFilter.getAttributes();
+        StudyViewFilter studyViewFilter = interceptedOncoKBDataCountFilter.getStudyViewFilter();
+        if (attributes.size() == 1) {
+            studyViewFilterUtil.removeSelfFromFilter(attributes.get(0).getAttributeId(), studyViewFilter);
+        }
+        List<SampleIdentifier> filteredSampleIdentifiers = studyViewFilterApplier.apply(studyViewFilter);
+
+        List<OncoKBDataCountItem> combinedResult = new ArrayList<>();
+        if (filteredSampleIdentifiers.isEmpty()) {
+            return new ResponseEntity<>(combinedResult, HttpStatus.OK);
+        }
+        List<String> studyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
+        List<Mutation> resultForSampleAttributes = mutationService.getMutationsInMultipleMolecularProfiles(molecularProfileService
+            .getFirstMutationProfileIds(studyIds, sampleIds), sampleIds, null, SUMMARY, 10000000, 0, null, null);
+        Set<OncoKBDataCount> set = new HashSet();
+        for(Mutation mutationProfile : resultForSampleAttributes) {
+          JSONObject jsonMutation = new JSONObject(mutationProfile.getAnnotation());
+          OncoKBDataCount oncoKBCancer = new OncoKBDataCount();
+          oncoKBCancer.setAttributeId("oncogenic");
+          oncoKBCancer.setValue(jsonMutation.getJSONObject("oncogenic").getString());
+          oncoKBCancer.setCount(oncoKBCancer.getCount()++);
```

Comment:
See comment in OncoKBDataCount. You cannot do the calculation without initialization